packages/vite-plugin-cloudflare/src/cloudflare-environment.ts

```diff
@@ -178,7 +158,30 @@ export function createCloudflareEnvironmentOptions(
 					external: [...cloudflareBuiltInModules],
 				},
 			},
-			webCompatible: true,
+			optimizeDeps: {
+				// Note: ssr pre-bundling is opt-in, and we need to enabled it by setting noDiscovery to false
+				noDiscovery: false,
+				exclude: [
+					...cloudflareBuiltInModules,
+					...builtinModules.concat(builtinModules.map((m) => `node:${m}`)),
+				],
+				esbuildOptions: {
+					platform: 'neutral',
```

Comment:
@petebacondarwin @jamesopstad I put `neutral` as that seemed the preference in chat? do you agree? or do you think that `browser` would be better?

Reply:
I think we should go with `neutral` as it seems more appropriate and then we can always change to `browser` later if needed.

Reply:
sounds good to me 🙂

Reply:
Also neutral adds in the fewest extra behaviours, which we can (mostly) add in manually if needed. The other modes require us to remove behaviours, which is generally more tricky.